torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1429,6 +1434,10 @@ def unpack_variables(args):
     'test_cdist',
 }
 
+EXCLUDE_METHOD_FOR_ATTRIBUTES = {
```

Comment:
probably no unless we change a lot of these lists. this condition checks if name is attribute https://github.com/pytorch/pytorch/blob/master/test/test_autograd.py#L4756-L4757 and then uses the method call in `getattr` which does not work for `real` and `imag` unlike other attributes. This might also be addressed in new autograd test suite cc. @mruberry

Reply:
Is the issue that they are attributes and not functions?
In that case, maybe you just want to update `add_test` to do `hasattr(self_variable, name) and callable(getattr(self_variable, name))` ?

Or better still, just access the attribute instead of calling the function there so that you can properly test the attribute access